### PR TITLE
Fix reporting reason in kubelet

### LINKED ISSUE #11007: Fix reporting of Reason in Kubelet
First of all, Reason (everywhere in the API) is intended to be a concise, single-world, CamelCase categorization of the status. However, Kubelet is producing phrases, which more properly belong in a Message field. For example:
https://github.com/GoogleCloudPlatform/kubernetes/blob/7bb9cdf660a00e231d9dcd1a7c9c5985399fb977/pkg/kubelet/dockertools/manager.go#L458
```
				Reason: fmt.Sprintf("Image: %s is ready, container is creating", image),
```

Secondly, important cases are somehow being dropped. Here's an example of a single-container pod, where the container must be Waiting, probably during image pull:
```bash
$ kubectl get pods
NAME          READY     STATUS    RESTARTS   AGE
hello-world   0/1       Running   0          7s
```
kubectl get pods shouldn't say "Running" for pods whose images aren't pulled.

Not sure if the reason is being clobbered or something else is happening:
https://github.com/GoogleCloudPlatform/kubernetes/blob/7bb9cdf660a00e231d9dcd1a7c9c5985399fb977/pkg/kubelet/dockertools/manager.go#L469

I wouldn't 100% rule out kubectl, but the logic there is pretty straightforward.

cc @andronat

### PR DESCRIPTION
**Pull request by jiangyaoguo**
fix #11007
1. Make reason field of StatusReport objects in kubelet in CamelCase format.
2. Add Message field for ContainerStateWaiting to describe detail about Reason.
3. Make reason field of Events in kubelet in CamelCase format.
4. Update swagger,deep-copy and so on.

### CODE REVIEW

File: pkg/kubelet/dockertools/manager.go [hunk 1]
```diff
@@ -502,11 +505,13 @@ func (dm *DockerManager) GetPodStatus(pod *api.Pod) (*api.PodStatus, error) {
...
 		if err == nil {
 			containerStatus.State.Waiting = &api.ContainerStateWaiting{
-				Reason: fmt.Sprintf("Image: %s is ready, container is creating", image),
+				Reason:  "ContainerCreating",
+				Message: fmt.Sprintf("Image: %s is ready, container is creating", image),
 			}
 		} else if err == docker.ErrNoSuchImage {
 			containerStatus.State.Waiting = &api.ContainerStateWaiting{
-				Reason: fmt.Sprintf("Image: %s is not ready on the node", image),
+				Reason:  "NoSuchImage",
```
Comment (bgrant0607) [inline pkg/kubelet/dockertools/manager.go]: Can we distinguish registry unresponsiveness/non-existence from image non-existence?

**Comment by bgrant0607** [comment]
Were the other instances of long reasons already fixed?

@dchen1107 please find a reviewer.

**Comment by bgrant0607** [comment]
cc @vishh regarding any monitoring systems that may be looking at these strings

File: pkg/kubelet/dockertools/manager.go (hunk 1 above)
Comment (dchen1107) [inline pkg/kubelet/dockertools/manager.go]: InspectImage only returns ErrNoSuchImage, and Pullmage does have both status codes: 200 for ErrNoSuchImage and 500 for unresponsive repository, but we didn't cache that result from different goroutine. Let's fix this one in a separate pr.

**Comment by dchen1107** [comment]
Quick grep found  more instances of long reasons which require fixes. 

@bgrant0607 I don't think any monitoring system today are looking at those Pod/ContainerStatus.

File: pkg/kubelet/dockertools/manager.go (hunk 1 above)
Comment (jiangyaoguo) [inline pkg/kubelet/dockertools/manager.go]: @bgrant0607 @dchen1107  I'll look into the reason cache and make another pr to make reason and message keep consistant with this.

File: pkg/kubelet/dockertools/manager.go [hunk 2]
```diff
@@ -332,17 +332,21 @@ func (dm *DockerManager) inspectContainer(dockerID, containerName, tPath string,
...
+		message := ""
 		// Note: An application might handle OOMKilled gracefully.
 		// In that case, the container is oom killed, but the exit
 		// code could be 0.
 		if inspectResult.State.OOMKilled {
-			reason = "OOM Killed"
+			reason = "OOMKilled"
 		} else {
-			reason = inspectResult.State.Error
+			reason = "ErrorOccur"
```
Comment (vishh) [inline pkg/kubelet/dockertools/manager.go]: nit: Can the reason be just `Error`?

**Comment by vishh** [comment]
Event reason changes LGTM.

File: pkg/kubelet/dockertools/manager.go (hunk 1 above)
Comment (dchen1107) [inline pkg/kubelet/dockertools/manager.go]: SGTM

**Comment by dchen1107** [comment]
LGTM. Thanks for the fix.

**Pull request merged**